language/diem-framework/modules/DiemTransactionPublishingOption.move

```diff
@@ -65,14 +65,13 @@ module DiemTransactionPublishingOption {
     /// Check if sender can execute script with `hash`
     public fun is_script_allowed(account: &signer, hash: &vector<u8>): bool {
         // DiemRoot can send any script
-        if (Roles::has_diem_root_role(account)) {
-            return true
-        };
+        if (Roles::has_diem_root_role(account)) return true;
 
         // No one except DiemRoot can send scripts when transactions are halted
-        if (transactions_halted()) {
-            return false
-        };
+        if (transactions_halted()) return false;
+
+        // An empty script hash implies a script function, which if invokable is allowed
```

Comment:
I find this a bit confusing--something like "the adapter passes an empty hash for script functions. All script functions are allowed" might be clearer?